### DISCUSSION TITLE
Projeto open-source para ajudar pessoas com depressão

### DISCUSSION THREAD

**Discussion by hectorgrecco** [frontendbr/forum#1137] [Discussão]
Pessoal, eu estou planejando um projeto full open-source com **Vue.js** com o seguinte propósito:

 - Pessoas depressivas contam suas histórias ou mandam alguma mensagem (seja anonimamente ou não), e qualquer pessoa pode respondê-la enviando textos ou frases de ajuda (seja de empatia e afins).

Sei consumir APIs, porém ainda não tenho conhecimento bacana para construir uma API com suas respectivas validações do zero. Há alguma ferramenta para criação de APIs utilizando PHP ou outra coisa via interface gráfica? Caso não haja, existe alguém por aqui que gostaria de contribuir com o projeto?

**Comment by ArtUssuy** [comment]
opa, não manjo de back end também, mas sou um bom front end, gostaria de contribuir, se possível.

**Comment by hectorgrecco** [comment]
@ArtUssuy show! É uma ideia que está no papel ainda, tinha pensado em "fazer tudo sozinho", mas aí me toquei que se tivesse a ajuda da comunidade seria bem mais bacana. Talvez role de fazer no Firebase, é algo a se pensar... Mas logo vou me organizar e criar um canal no Slack ou algo parecido, não tenho experiência com projetos open-source, então seria uma oportunidade ótima. Obrigado pelo apoio =)

**Comment by kivervinicius** [comment]
@hectorgrecco  da uma olhada é muito bom https://github.com/strapi/strapi

**Comment by hectorgrecco** [comment]
Ops, @LFeh  agora que eu vi que o FrontEndBr tem um repositório só pra pedir ajuda em projetos open-source. Foi mal! kakaka

**Comment by Raffesmind** [comment]
Tenta trabalhar com o firebase ou com o graphcool pode te dar a api em json
para você trabalhar visualmente

Em qua, 4 de jul de 2018 10:27, Hector Grecco <notifications@github.com>
escreveu:

**Comment by hectorgrecco** [comment]
Valeu pelas dicas, @kivervinicius e @Raffesmind =)

**Comment by pcesarteixeira** [comment]
Projeto maneiro, tenho interesse em ajudar, pode me contatar que tamo junto. Sou bom em elaborar layouts e desenvolver eles, alem de trabalhar com VueJS também. ;)

**Comment by barbier** [comment]
Olá!
Indo um pouco contra a maré, existe um projeto do tipo chamado [If me](https://www.if-me.org/) que também é open-source. Talvez seja melhor ver com eles se é possível adaptar a plataforma ao público brasileiro?

**Comment by hectorgrecco** [comment]
Criei um canal no Slack, pra quem tiver interesse:

https://join.slack.com/t/projeto-opensource/shared_invite/enQtMzkzOTEzMDA3Mjg2LTc5MGM0YjhjZDA0YjgzNTI1MzJlZTgyMDc1YzA4MWMyNDlkMTczNGIyZDA1NGVkOWMxYTUwMzQ4NGNmMDc4ZjM

**Comment by luizbills** [comment]
recomendo o https://getdirectus.com/ para API ou usar a REST API do WordPress.

**Comment by synga** [comment]
Cara demais isso, acho importantíssimo usar tecnologia pra algo assim, tantas possibilidades e nenhuma ferramenta ainda, parabéns pela iniciativa, vou entrar no grupo assim que possível.
Creio que a parte técnica é a mais fácil de superar, já deram a ideia do Firebase e acho uma boa, trabalho bastante e é fácil (inclusive com Vue), te da muitas ferramentas legais pra trampar. Então é só aprender a trabalhar com NoSQL pra desvendar o DB deles e já era, o resto você tira de letra;
Uma outra dica seria fazer disso um PWA, acho que pra um projeto assim ter como enviar pushs é uma otima, umas mensagens legais e etc.
Mas o mais importante mesmo é o core essa ideia ai, chama umas 2 ou 3 pessoas que estudem psicologia pra participar do projeto, ter o nome nisso ai e tudo mais, pra poder saber qual a melhor forma de lidar com isso. Digo isso por que abrir comentários para o publico SEMPRE vai ter gente que vai querer fazer mal, SEMPRE, então tem que tomar um cuidado por que nunca se sabe o que pode ser um gatilho pra uma pessoa com depressão, qual o nível de depressão dela e o que ela pode tentar se disparar um gatilho. Acho que a execução do projeto vai ser fácil se você desvendar todas essas coisas ai.

**Comment by hectorgrecco** [comment]
Valeu demais, @synga ! Igual você falou, tá fácil o projeto sair do papel, e pelos comentários só tem gente boa aqui disposta a ajudar no projeto =)

**Comment by felipefialho** [comment]
@hectorgrecco Relaxa, dado o caráter nobre dessa idéia, pode manter aqui sem problemas :)

**Comment by ericvvf** [comment]
@hectorgrecco que ideia nobre! Recentemente, enfrentei uma barra com um familiar bem próximo, e acabei me envolvendo de cabeça no problema. É um processo muito difícil e desgastante. Hoje, graças a Deus, e a nossos esforços, temos a situação mais controlada. Durante o processo, nos momentos mais difíceis, dentro de mim, sempre vinha uma vontade de ajudar outras pessoas, com meu trabalho. Pensei em falar com a psicóloga da faculdade que estudo, e em criar algo do tipo. Porém, neste meio tempo, minha esposa engravidou e meu tempo ficou muito curto. Gostaria muito de ajudar, meu foco é mais o back end (Django, Node, Laravel, Yii), mas também arranho no AngularJS e Angular, e muito pouco, ainda, de React. Não sei se terei tempo, devido ao nascimento do meu filho, mas irei acompanhar (vou entrar no grupo do slack) o desenrolar, e quem sabe, se precisarem ainda, eu possa ajudar em algo.

E reforço a ideia do @synga: procura parceria com pessoas da área de psicologia e/ou psiquiatria. 

Um abraço, e que sua nobre ideia ajude muitas pessoas!

**Comment by hectorgrecco** [comment]
@ericOnda Fala, meu velho! Beleza? Então, minha namorada tirou a própria vida há pouco mais de 2 meses por conta da depressão, e uns dias atrás resolvi colocar a ideia em prática pra ajudar outras pessoas que talvez passem pela mesma barra. Já entrei em contato com um pessoal da psicologia, psiquiatria e até serviço social, pra expandir as funcionalidades da plataforma com um blog, algo assim. O projeto vai andar.. de verdade!
Valeu!!!

**Comment by felipefialho** [comment]
@hectorgrecco, fiquei realmente chocado em ler isso. Muita força mano.

Acabei de divulgar essa issue no [Twitter](https://twitter.com/frontendbr/status/1014666997865615361?s=19)

**Comment by crisgon** [comment]
Parabéns pela ideia!! Gostei muito e quero ajudar!!!

**Comment by ericvvf** [comment]
Poxa vida, @hectorgrecco ! Que dureza, irmão! Desejo, de coração, que Deus lhe conceda forças e sabedoria. Em alguns momentos, é só o que eu desejo. Este projeto vai andar, e vai ajudar muita gente! Força!!

**Comment by hectorgrecco** [comment]
Muito obrigado pessoal! O grupo no Slack tá rolando e já tem um pessoal bacana lá! Bora?
https://bit.ly/2zeH7Ek

**Comment by rubenmarcus** [comment]
po podia ser um app também e tipo uma rede social voltada a pessoas com depressão

**Comment by rubenmarcus** [comment]
qual o endereço do repositório?

**Comment by hectorgrecco** [comment]
@rubenmarcus Ainda não tem nada pronto, só mais ou menos o Backlog do Produto. Se liga no que postei no grupo do slack:

No mais, penso que as funcionalidades serão essas:
- A pessoa se cadastra via e-mail.
- Terá um painelzinho pra ela enviar sua história, ou mensagem. Na mesma tela, ela escolhe se aquela história será anônima ou não
- Dentro desse feed de mensagens das pessoas, outras poderão logar-se e enviar mensagens de carinho, ajuda, músicas, vídeos, qualquer outra coisa (porém, essas mensagens só aparecerão pra pessoa se for aprovada por algum administrador... A minha é que gente da psicologia, psiquiatria, que seja moderem esses comentários. A moderação serve mais pra evitar haters/trolls pra evitar que a pessoa fique mais pra baixo ainda. Talvez, para minimizar esses haters, rola de pedir CPF no cadastro)
- Pensei num serviço dentro da plataforma que lista os consultórios de psicologia e psiquiatria gratuitos próximo da pessoa utilizando Geo-localização. (Os dados abertos vem da API oficial do TCU: https://bit.ly/2u8fbvY) (edited)
- E por último, um mini blog pra ajudar o pessoal, com dicas, dicas pra familiares da pessoa com depressão
Vai ser uma Single-Page Application (SPA) + PWA (Progressive Web App). Depois mando o conceito e uns tutoriais pra quem não manjar muito

**Comment by arthureducadoresnolinux** [comment]
Rapaz, eu adoraria ajudar no seu projeto. É uma ideia ótima!
